lms/static/js/search/views/search_list_view.js

```diff
@@ -57,7 +57,7 @@ define([
                 var item = new SearchItemView({ model: result });
                 return item.render().el;
             });
-            this.$el.find('.search-results').append(items);
+            this.$el.find('.search-results').html(items);
```

Comment:
Connected to the same issue as below. Had to be done because collection needs to be aware of all of the elements in list view, not just last returned set (appended one). That's why it is redrawn completely.